tapdb/ignore_tree.go

```diff
@@ -171,98 +158,121 @@ func (it *IgnoreUniverseTree) Sum(ctx context.Context,
+}
 
-	var tuples []*universe.IgnoreTuple
-
-	readTx := NewBaseUniverseReadTx()
-	txErr := it.db.ExecTx(ctx, &readTx, func(db BaseUniverseStore) error {
-		// To list all the tuples, we'll just query for all the universe
-		// leaves for this namespace. The namespace is derived from the
-		// group key, and the proof type, which in this case is ignore.
-		universeLeaves, err := db.QueryUniverseLeaves(
-			ctx, UniverseLeafQuery{
-				Namespace: namespace,
-			},
-		)
-		if err != nil {
-			return err
+// queryIgnoreLeaves retrieves UniverseLeaf records based on IgnoreTuple
+// criteria.
+func queryIgnoreLeaves(ctx context.Context, dbtx BaseUniverseStore,
+	spec asset.Specifier,
+	tuples ...universe.IgnoreTuple) ([]UniverseLeaf, error) {
+
+	uniNamespace, err := specifierToIdentifier(
+		spec, universe.ProofTypeIgnore,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error deriving identifier: %w", err)
+	}
+
+	var allLeaves []UniverseLeaf
+	for _, queryTuple := range tuples {
+		// Create a leaf query for this specific key.
+		//
+		// TODO(roasbeef): need a more specific key here?
+		//   * also add outpoint?
+		scriptKey := queryTuple.ScriptKey
+		leafQuery := UniverseLeafQuery{
+			ScriptKeyBytes: scriptKey.SchnorrSerialized(),
+			Namespace:      uniNamespace.String(),
 		}
 
-		for _, leaf := range universeLeaves {
-			leafBytes := leaf.GenesisProof
+		leaves, err := dbtx.QueryUniverseLeaves(ctx, leafQuery)
 
-			tuple, err := universe.DecodeSignedIgnoreTuple(
-				leafBytes,
-			)
-			if err != nil {
-				return fmt.Errorf("error decoding tuple: "+
-					"%w", err)
-			}
+		// We'll continue on to the next query if no leaves are found
+		// for this query.
+		if errors.Is(err, sql.ErrNoRows) {
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error querying leaf for tuple "+
+				"(script_key=%x): %w", leafQuery.ScriptKeyBytes,
+				err)
+		}
 
-			tuples = append(tuples, &tuple.IgnoreTuple.Val)
+		// Since the query is specific, we expect at most one leaf.
+		if len(leaves) > 0 {
+			allLeaves = append(allLeaves, leaves[0])
 		}
+	}
 
-		return nil
-	})
+	// Return sql.ErrNoRows if no leaves were found across all tuples.
+	if len(allLeaves) == 0 {
+		return nil, sql.ErrNoRows
+	}
 
-	if txErr != nil {
-		return lfn.Err[[]*universe.IgnoreTuple](txErr)
+	return allLeaves, nil
+}
+
+// decodeAndBuildAuthIgnoreTuple decodes the raw leaf, reconstructs the key,
+// and builds the AuthenticatedIgnoreTuple.
+func decodeAndBuildAuthIgnoreTuple(dbLeaf UniverseLeaf,
+) (*universe.SignedIgnoreTuple, uniKey, error) {
+
+	signedTuple, err := universe.DecodeSignedIgnoreTuple(
+		dbLeaf.GenesisProof,
+	)
+	if err != nil {
+		return nil, uniKey{}, fmt.Errorf("error decoding tuple: %w",
+			err)
 	}
 
-	return lfn.Ok(tuples)
+	return &signedTuple, signedTuple.UniverseKey(), nil
+}
+
+// buildAuthIgnoreTuple constructs the final AuthenticatedIgnoreTuple.
+func buildAuthIgnoreTuple(decodedLeaf *universe.SignedIgnoreTuple,
```

Comment:
Perhaps this can become a `universe.AuthenticatedIgnoreTuple` constructor called `NewAuthIgnoreTuple`

Reply:
BTW, I'm getting an error in my IDE because `decodedLeaf` is a pointer where as field `SignedIgnoreTuple` is not.

Reply:
> BTW, I'm getting an error in my IDE because decodedLeaf is a pointer where as field SignedIgnoreTuple is not.

This happened post rebase, should be resolved in the latest push.

Reply:
Done!